File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    var answer = '';
    var tempArr = []; // 임시배열 선언
    
    // 1부터 food의 길이만큼 반복하면서
    for (i = 1; i < food.length; i++) {
        // 임시배열에 food의 필요한 갯수를 나눈 몫을 저장해준다.
        tempArr.push(Math.floor(food[i]/2));
    }
    
    // tempArr의 배열을 탐색하면서
    for (i = 0; i < tempArr.length; i++) {
        // 만약 tempArr의 i번 인덱스의 숫자가 0보다 크다면
        if (tempArr[i] > 0) {
            // tempArr의 i번 인덱스의 숫자만큼 반복하면서
            for (j = 0; j < tempArr[i]; j++) {
                answer += (i+1); // answer에 i번인덱스 + 1의 숫자를 더해준다
            }
        }
    }
    // 모든 반복이 종료되었다면 answer 문자열에 0과 반전시킨 answer 문자열을 더해준다
    answer += "0" + answer.split("").reverse().join("");
    
    return answer;
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    answer = ''
    tempArr = [] # 사용할 음식 수를 담아줄 임시배열 선언
    
    # food[1~3]의 몫을 tempArr에 담아준다
    for i in range(1, len(food)):
        tempArr.append(food[i]//2)
    
    # 대회에 사용될 음식만큼 반복하면서
    for i in range(len(tempArr)):
        if tempArr[i] > 0: # 만약 i가 0보다 큰 숫자일 경우
            for j in range(tempArr[i]): # tempArr에 할당된 갯수만큼
                # answer 에 음식을 더해준다 (0번 인덱스부터 1번 음식이라 i + 1)
                answer += str(i + 1) 
    
    # 정배열이 끝나면 0과 정배열의 완성된 문자열을 뒤집어 추가해준다.
    answer += "0" + answer[::-1]
    
    return answer